### PR TITLE
fix(tags): handle non-semvers tag

### LINKED ISSUE #321: ingressclass template causes chart install to fail when using non-semver values in `image.tag`
I encountered this issue when trying to force the helm chart to use the `latest` image tag when testing a new feature. This bug cannot be worked around because the logic is always going to evaluate the SemVer in the following code (even if `.Values.ingressClass.enabled` evaluates to `false`):

https://github.com/traefik/traefik-helm-chart/blob/master/traefik/templates/ingressclass.yaml#L1

Do we need this logic check here? Is it not enough to assume the operator is enabling `ingressClass` when they understand their cluster has the required APIs and support? See related issue - https://github.com/traefik/traefik-helm-chart/issues/309

values.yaml:

```yaml
        image:
          name: traefik/traefik
          tag: latest
          pullPolicy: Never

```

Result:

```
can't install chart `traefik`: template: traefik/templates/ingressclass.yaml:1:41: executing \"traefik/templates/ingressclass.yaml\" at <semverCompare \">=2.3.0\" (default .Chart.AppVersion .Values.image.tag)>: error calling semverCompare: Invalid Semantic Version"
```

### PR DESCRIPTION
**Pull request by faust64**
Ensures IngressClass installation check would properly handle deployment for traefik `latest` tag.

Fixes #321

### CODE REVIEW

**Comment by SantoDE** [comment]
Hey @faust64,

can you bump the version again please? I would do so, but looks like you did not allow edits from maintainers :)

File: traefik/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: traefik
 description: A Traefik based Kubernetes ingress controller
 type: application
-version: 9.14.3
+version: 9.14.4
```
Comment (SantoDE) [inline traefik/Chart.yaml:5]: Needs to be 9.15.5

**Review by SantoDE — CHANGES_REQUESTED** [review]
See above

**Review by SantoDE — COMMENTED** [review]
LGTM. Thanks!

**Review by SantoDE — APPROVED** [review]
LGTM

**Comment by SantoDE** [comment]
Hey @faust64,

can you please provide maintainers with permissions to do edits on that PR? That would help a lot.

Thanks!

**Comment by faust64** [comment]
I'm not sure I can. AFAIU, there should be a button on under the Unsubscribe one, in that pull request ( https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork )

I can't find it, not sure why:
unsubscribe-btn

let me know if there's anything I can do.

**Comment by SantoDE** [comment]
Somehow, I cant push to your branch. No matter what I try.

Can you open the PR (check the box) to allow edits from maintainer? :)

**Comment by faust64** [comment]
Sorry, as stated earlier, I don't see that checkbox. This seems to be due to my forking to my organization instead of my own account. I can re-open my PRs, forking again, if there's no way around this.

**Comment by faust64** [comment]
see https://github.com/traefik/traefik-helm-chart/pull/384

**Pull request closed without merging**